### DISCUSSION TITLE
Feature request: add optional sensor for grid cost

### DISCUSSION THREAD

**Discussion by ragg987** [slipx06/sunsynk-power-flow-card#496] [Ideas]
Hi, it would be nice if you can add the ability to bring in grid cost as optional sensor. I use the sunsynk card on my HA landing page as my primary power and energy dashboard, however I keep having to refer elsewhere for the cost aspect.

Suggested layout to the sensors:
![image](https://github.com/user-attachments/assets/ab2d31b7-20e8-4866-97ad-ca2d21bcecff)

**Comment by slipx06** [comment]
There are two sensors that show you the current buy and sell cost. You could use those to display your cumulative cost. The only disadvantage to this is that it will only display one of these at a given time depending on whether you are importing or exporting. the sensors are

```
energy_cost_buy: | Optional |   | Sensor that provides current buy energy cost per kWh
energy_cost_sell: | Optional |   | Sensor that provides current sell energy cost per kWh
```

**Reply by ragg987** [reply]
Thanks - I am already using these to show the current cost (current sell is showing as 0 in my image). The cumulative costs would be an additional sensors, I appreciate this is not the primary function of this card hence putting it out there as a feature request. A flexible option would be to add additional "extra"sensors to use as the user deems fit - in the same way as you have provided with the load.

```
essential_load1_extra:
```
